jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -199,16 +201,102 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
+                                        diagnostic2 = createDiagnostic(onMessageTextUsed, unit,
+                                                WebSocketConstants.DIAGNOSTIC_ON_MESSAGE_DUPLICATE_METHOD,
+                                                WebSocketConstants.DIAGNOSTIC_CODE_ON_MESSAGE_DUPLICATE_METHOD);
+                                        diagnostics.add(diagnostic1);
+                                        diagnostics.add(diagnostic2);
+                                    }
+                                    onMessageTextUsed = annotation;
+                                    break;
+                                case BINARY:
+                                    if (onMessageBinaryUsed != null) {
+                                        diagnostic1 = createDiagnostic(annotation, unit,
+                                                WebSocketConstants.DIAGNOSTIC_ON_MESSAGE_DUPLICATE_METHOD,
+                                                WebSocketConstants.DIAGNOSTIC_CODE_ON_MESSAGE_DUPLICATE_METHOD);
+                                        diagnostic2 = createDiagnostic(onMessageBinaryUsed, unit,
+                                                WebSocketConstants.DIAGNOSTIC_ON_MESSAGE_DUPLICATE_METHOD,
+                                                WebSocketConstants.DIAGNOSTIC_CODE_ON_MESSAGE_DUPLICATE_METHOD);
+                                        diagnostics.add(diagnostic1);
+                                        diagnostics.add(diagnostic2);
+                                    }
+                                    onMessageBinaryUsed = annotation;
+                                    break;
+                                case PONG:
+                                    if (onMessagePongUsed != null) {
+                                        diagnostic1 = createDiagnostic(annotation, unit,
+                                                WebSocketConstants.DIAGNOSTIC_ON_MESSAGE_DUPLICATE_METHOD,
+                                                WebSocketConstants.DIAGNOSTIC_CODE_ON_MESSAGE_DUPLICATE_METHOD);
+                                        diagnostic2 = createDiagnostic(onMessagePongUsed, unit,
+                                                WebSocketConstants.DIAGNOSTIC_ON_MESSAGE_DUPLICATE_METHOD,
+                                                WebSocketConstants.DIAGNOSTIC_CODE_ON_MESSAGE_DUPLICATE_METHOD);
+                                        diagnostics.add(diagnostic1);
+                                        diagnostics.add(diagnostic2);
+                                    }
+                                    onMessagePongUsed = annotation;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Create an error diagnostic if a ServerEndpoint annotation does not follow the
```

Comment:
Please add client endpoint to the comment and also explain the rule a bit.

Reply:
I don't know why I added ClientEndpoint to this method as it's not in the spec but this method was part of a previous PR. Regardless, I've removed the client endpoint check and added a rule explanation.